src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -12559,6 +12557,8 @@ void Unit::RemoveCharmedBy(Unit* charmer)
         }
         player->SetClientControl(this, true);
     }
+    
+    EngageWithTarget(charmer);
```

Comment:
But that was my point - on retail if a creature/boss charms a player who is the only entry on the threat list, it evades.